pennylane/templates/layers/particle_conserving_u1.py

```diff
@@ -0,0 +1,267 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the hardware efficient ``ParticleConservingU1`` template.
+"""
+import numpy as np
+
+import pennylane as qml
+
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.templates.decorator import template
+from pennylane.ops import CNOT, CRot, PhaseShift, CZ
+from pennylane.templates.utils import (
+    check_shape,
+    check_type,
+    get_shape,
+)
+from pennylane.wires import Wires
+
+
+def decompose_ua(phi, wires=None):
+    r"""Implement the circuit decomposing the controlled application of the unitary
+    :math:`U_A(\phi)`
+
+    .. math::
+
+        U_A(\phi) = \left(\begin{array}{cc} 0 & e^{-i\phi} \\ e^{-i\phi} & 0 \\ \end{array}\right)
+
+    in terms of the quantum operations supported by PennyLane.
+
+    :math:`U_A(\phi)` is used in `arXiv:1805.04340 <https://arxiv.org/abs/1805.04340>`_),
+    to define two-qubit exchange gates required to build particle-conserving
+    VQE ansatze for Quantum Chemistry simulations. See :func:`~.ParticleConservingU1`.
+
+    :math:`U_A(\phi)` is expressed in terms of ``PhaseShift``, ``Rot`` and ``PauliZ`` operations
+    :math:`U_A(\phi) = R_\phi(-2\phi) R(-\phi, \pi, \phi) \sigma_z`.
+
+    Args:
+        phi (float): angle :math:`\phi` defining the unitary :math:`U_A(\phi)`
+        wires (list[Wires]): the wires ``n`` and ``m`` the circuit acts on
```

Comment:
While templates are supposed to understand the `Wire` class (and therefore this input is valid), the class is not exposed to users, who will usually just use lists. 

Maybe use the type description from other templates (`Iterable or Wires`)? It's not perfect I know.

Reply:
Oh I see, this is a helper function. Ignore my comment :)